### PR TITLE
fix: Add char limits for addPerson fields

### PR DESCRIPTION
**Pull request by freshcabbage123**
## What type of PR is this? (check all applicable)

-   [ ] Refactor
-   [ ] Feature
-   [X] Bug Fix
-   [ ] Optimization
-   [ ] Documentation Update

## Description

-   Add char limits for addPerson fields, noting that email's limit is set at 6 as AB3's requirement for domain name to be at least 2 char.

## Related Tickets & Documents

-   Related Issue #144 

## Added/updated tests?

_We encourage you to keep the code coverage percentage at 75% and above for new changes._

-   [X] Yes
-   [ ] No, and this is why: _please replace this line with details on why tests
        have not been included_
-   [ ] No, there are UI changes that I'm unable to cover.
-   [ ] I need help with writing tests

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddPersonCommandParser.java [hunk 1]
```diff
@@ -45,6 +45,14 @@ public AddPersonCommand parse(String args) throws ParseException {
...
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
+        // tag's limit is 30 characters, but we set it to 32 to account for the square brackets that encapsulate it
+        if (name.toString().isEmpty() || name.toString().length() > 100
+                || phone.toString().length() < 3 || phone.toString().length() > 20
+                || email.toString().length() < 6 || address.toString().length() > 200
+                || tagList.stream().anyMatch(tag -> tag.toString().length() > 32)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddPersonCommand.MESSAGE_USAGE));
+        }
+
```
Comment (jiakai-17) [inline src/main/java/seedu/address/logic/parser/AddPersonCommandParser.java:55]: Would it be better if you extract these checks into the constructor of each Object? 

That way, it is easier to maintain, plus we can throw a more detailed error message (eg: Address too long vs Invalid Command format)

Comment (freshcabbage123) [reply src/main/java/seedu/address/logic/parser/AddPersonCommandParser.java:55]: @jiakai-17 i could do that. But would throwing a more detailed error message be counted as adding a new feature? I was under the impression that we could add limitations to the length of the fields so long as the error message was the same i.e. invalid command error msg

Comment (jiakai-17) [reply src/main/java/seedu/address/logic/parser/AddPersonCommandParser.java:55]: @freshcabbage123 Okay, I guess the feature freeze limits us from writing more descriptive error messages. Do also remember to add a point about more descriptive error messages in the Planned Enhancements section.

**Comment by HugeNoob** [comment]
> seems like a bopes

Are you planning to mention this in the DG or UG?

**Pull request closed without merging**